tt_eager/tt_dnn/op_library/composite/composite_ops.cpp

```diff
@@ -1401,8 +1400,17 @@ Tensor hypot(const Tensor& input_a, const Tensor& input_b, const MemoryConfig& o
 
 Tensor _scatter(const Tensor& input_a, const Tensor& input_b, const MemoryConfig& output_mem_config) {
     const Shape start_index = {0, 0, 0, 0};
-    Tensor index = pad(ones_like(input_a, output_mem_config), input_b.get_legacy_shape(), start_index, 0);
-    Tensor temp_a = pad(input_a, input_b.get_legacy_shape(), start_index, 0);
+    ttnn::Tensor input_tensor_4D = ttnn::unsqueeze_to_4D(input_a);
+    auto input_shape_with_tile_padding = input_tensor_4D.get_shape().with_tile_padding();
+    auto output_padded_shape = input_b.legacy_shape();
+    std::vector<std::pair<uint32_t, uint32_t>> padding(4);
+    for(size_t i = 0; i < padding.size(); i++) {
+        padding[i] = {0, output_padded_shape[i] - input_shape_with_tile_padding[i]};
+    }
```

Comment:
Why did tt_dnn version not require this before and ttnn requires? Maybe we can place that inside the op itself?

Reply:
I remember adding unsqueeze_to_4d inside ttnn pad 🤔

Reply:
Ya its just a diff way of thinking about the problem, either to think of it as the output shape we want (tt_dnn) way, or to think of padding explicitly (ttnn and PyTorch) way. I can add another function signature such that both tt_dnn way and ttnn/pytorch way works out of the box

Reply:
also ya you needed to add that inside ttnn pad because you were calling the tt_lib device code which expects the tt_dnn format of padding. Once tt_lib accepts padding just one way then we won't need that.